gammapy/utils/tests/test_random.py

```diff
@@ -2,14 +2,55 @@
 from __future__ import print_function, division
 import numpy as np
 from numpy.testing import assert_allclose
+from astropy.coordinates import Angle, Longitude, Latitude
+from astropy.tests.helper import assert_quantity_allclose
 from ..random import sample_sphere, sample_powerlaw, sample_sphere_distance
 
 
 def test_sample_sphere():
+
+    # test general case
     np.random.seed(0)
     lon, lat = sample_sphere(size=2)
-    assert_allclose(lon, [3.44829694, 4.49366732])
-    assert_allclose(lat, [0.20700192, 0.08988736])
+    assert_quantity_allclose(lon, Longitude([3.44829694, 4.49366732], 'radian'))
+    assert_quantity_allclose(lat, Latitude([0.20700192, 0.08988736], 'radian'))
+
+    # test specify a limited range
+    lon_range = Longitude([40., 45.], 'degree')
+    lat_range = Latitude([10., 15.], 'degree')
+    lon, lat = sample_sphere(size=10,
+                             lon_range=lon_range,
+                             lat_range=lat_range)
+    assert ((lon_range[0] <= lon) & (lon < lon_range[1])).all()
+    assert ((lat_range[0] <= lat) & (lat < lat_range[1])).all()
+
+    # test lon within (-180, 180) deg range
+    lon_range = Longitude([-40., 0.], 'degree', wrap_angle=Angle(180., 'degree'))
+    lon, lat = sample_sphere(size=10, lon_range=lon_range)
+    assert ((lon_range[0] <= lon) & (lon < lon_range[1])).all()
+    lat_range = Latitude([-90., 90.], 'degree')
+    assert ((lat_range[0] <= lat) & (lat < lat_range[1])).all()
+
+    # test lon range explicitly (0, 360) deg
+    epsilon = 1.e-8
+    lon_range = Longitude([0., 360.-epsilon], 'degree')
```

Comment:
@cdeil This way of defining the 360 deg boundary is necessary to avoid the `Longitude` wrapper to convert it to `0 deg`. I'm not sure this is better than just using `Angle` objects, even if the angles might need to be wrapped. BTW, I'm not even sure if we have to wrap them at all. Won't the method just work fine, provided the angle ranges are not broader than 360 deg? (I haven't tested it).